fuel-tx/src/transaction/validity/error.rs

```diff
@@ -90,6 +90,10 @@ pub enum ValidityError {
     TransactionOutputContainsContractCreated {
         index: usize,
     },
+    /// The output contains a `Output::Coin` which is not allowed.
+    TransactionOutputContainsCoin {
```

Comment:
It is not described anywhere, but for backward compatibility, we need to add all new variants of errors at the end. I want to add a test later to ensure this behaviour.

But for now could you please add all new variants to the end?=)

Reply:
https://github.com/FuelLabs/fuel-vm/issues/797

Reply:
Also, we want to allow at least `Output::Coin` for the base asset. I think we don't need this error and this check. The `TransactionInputContainsNonBaseAssetId` error should be enough.

Reply:
ac2776ac changes the check to do that